employee_o_matic_4000.rb

```diff
@@ -0,0 +1,175 @@
+class Employee
+    attr_accessor :full_name
+    attr_accessor :id
+
+    def initialize(full_name, id)
+        @full_name = full_name
+	@id = id
+    end
+
+    def last_name
+        @full_name.split(' ', 2).last
+    end
+
+    def first_name
+        @full_name.split(' ', 2).first
+    end
+end
+
+class Programmer < Employee
+    attr_accessor :languages
+
+    def initialize(full_name, id, languages)
+        super(full_name, id)
+	@languages = languages.split(', ')
+    end
+end
+
+class OfficeManager < Employee
+    attr_accessor :office
+
+    def initialize(full_name, id, office)
+        super(full_name, id)
+        @office = "(#{office})"
+    end
+
+    def off
+        @office.downcase[1]
+    end
+end
+
+# Here we are declaring methods
+
+def add_employee(employees)
+    puts '[Add an employee]'
+    print 'Full name: '
+    full_name = gets.chomp
+    print 'ID: '
+    id = gets.chomp
+
+    print 'Is the person an [e]mployee, [p]rogrammer or an [o]ffice manager? '
+    action = get_action
+    case action
+    when 'e' then employee = Employee.new(full_name, id)
+    when 'p' then
+        print 'Programming languages: '
+        languages = gets.chomp	
+        employee = Programmer.new(full_name, id, languages)
+    when 'o' then
+        print 'Office: '
+        office = gets.chomp
+	employee = OfficeManager.new(full_name, id, office)
+    else
+        puts 'That wasn\'t an instruction!'
+	employee = Employee.new(full_name, id)
+    end
+
+    employees << employee
+end
+
+def edit_employee(employees)
+    puts '[Edit an employee]'
+    print 'ID of the employee you want to edit: '
+    id = gets.chomp
+    employees.each do |employee|
```

Comment:
Take a look at `Array#find` it's much clearer what it does than looping through the whole array.